Classes/Hooks/DataHandler.php

```diff
@@ -387,5 +349,53 @@ public function processCmdmap_postProcess($command, $table, $id)
                 }
             }
         }
+        if (
+            $command === 'delete'
+            && $table == 'tx_dlf_solrcores'
+        ) {
+            // Delete core from Apache Solr as well.
+            $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)
+                ->getQueryBuilderForTable('tx_dlf_solrcores');
+            // Record in "tx_dlf_solrcores" is already deleted at this point.
+            $queryBuilder
+                ->getRestrictions()
+                ->removeByType(DeletedRestriction::class);
+
+            $result = $queryBuilder
+                ->select(
+                    'tx_dlf_solrcores.index_name AS core'
+                )
+                ->from('tx_dlf_solrcores')
+                ->where($queryBuilder->expr()->eq('tx_dlf_solrcores.uid', intval($id)))
+                ->setMaxResults(1)
+                ->execute();
+
+            $allResults = $result->fetchAll();
+
+            if (count($allResults) == 1) {
+                $resArray = $allResults[0];
+                // Establish Solr connection.
+                $solr = Solr::getInstance();
+                if ($solr->ready) {
+                    // Delete Solr core.
+                    $query = $solr->service->createCoreAdmin();
+                    $action = $query->createUnload();
+                    $action->setCore($resArray['core']);
+                    $action->setDeleteDataDir(true);
```

Comment:
Honeystly. I wouldn't add this feature. This can destroy the indexing work of several months by a wrong click in the backend. Or you move installation to another system and do not think about chaning the Solr configuration.
Deleting a solr core is a super admin job. So the person knows what he/she is doing and can do this even better in the Solr admin page.
If we add this feature to TYPO3, we need at least one of the following:
1. Possibility to disable this feature
2. Better do it in some backend-module
3. Add a dialog to confirm the deletion

Reply:
I've added the first option: In order to automatically delete a Solr core when deleting the corresponding TYPO3 record, you have to set `solrAllowCoreDelete` to `true` in extension configuration (default is `false` to prevent accidents).

Reply:
Perfect. This works as expected. The core is only deleted if this option is activated.